addons/mail/static/src/models/thread.js

```diff
@@ -1539,11 +1540,9 @@ registerModel({
          * @returns {Throttle}
          */
         _computeThrottleNotifyCurrentPartnerTypingStatus() {
-            return throttle(
-                this.messaging,
-                ({ isTyping }) => this._notifyCurrentPartnerTypingStatus({ isTyping }),
-                2.5 * 1000
-            );
+            return insertAndReplace({
+                func: ({ isTyping }) => this._notifyCurrentPartnerTypingStatus({ isTyping }),
+            });
```

Comment:
If this is no longer a throttle version that is returned this field/compute is no longer necessary, `_notifyCurrentPartnerTypingStatus` could be used in place of `throttleNotifyCurrentPartnerTypingStatus`.

Reply:
Throttle should know what to call depending on its identifying fields?

Reply:
I kept like this right now in order to keep the same technical tests of Throttle, to ensure the exact same behaviour as a `Model` as when it was an `Utils`.

But I completely agree this is much better as more declarative way like in a computed field or just immediately handle in the "time out" part of code of `Throttle`.
Just keep in mind that if we do this, we need to make further changes to test, e.g. override `Throttle` methods for `QUnit` usages.